.circleci/config.yml

```diff
@@ -0,0 +1,114 @@
+version: 2.1
+orbs:
+  node: circleci/node@5.0.2
+jobs:
+  build:
+    executor: node/default
+    steps:
+      - checkout
+      - node/install-packages:
+          pkg-manager: npm
+      - run:
+          command: sudo npm install -g ts-node
+          name: Install ts-node
+      - run:
+          command: ts-node -v
+          name: Check ts-node version to make sure it installed
+      - persist_to_workspace:
+          root: .
+          paths:
+            - '*'
+
+  test:
+    executor: node/default
+    steps:
+      - attach_workspace:
+          at: .
+      - checkout
+      - run:
+          command: npm run test
+          name: run tests
+
+  format:
+    executor: node/default
+    steps:
+      - attach_workspace:
+          at: .
+      - checkout
+      - run:
+          name: format
+          command: npm run format
+
+  lint:
+    executor: node/default
+    steps:
+      - attach_workspace:
+          at: .
+      - checkout
+      - run:
+          name: lint
+          command: npm run lint
```

Comment:
Look at how this was done in the solana trader API FFI server, the `npm run` commands fix the errors, and not even all of them.

What you need is something to show the errors.

Reply:
We shouldn't temporarily fix the code in the CircleCI version

Reply:
no they don't fail tests and they do not update the repo with any formatting changes. If it is passed a warning level, the format and lint checks will fail when run

Reply:
they might still emit a non-zero exit code, which is what's needed for this step to fail. I don't know the specifics of this though.

Reply:
I changed the linter to use flag called `--max-warnings` as done in the go sdk. this now forces an error code on lint error so the job fails if there is a lint error/warning. tested with comments, type errors